aten/src/ATen/native/mps/operations/MaxUnpooling.mm

```diff
@@ -0,0 +1,141 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/native/mps/MPSGraphVenturaOps.h>
+#include <ATen/native/mps/OperationUtils.h>
+
+#include <ATen/ops/max_unpool2d.h>
+#include <ATen/ops/max_unpool2d_native.h>
+
+namespace at::native {
+namespace mps {
+struct CachedGraph : public MPSCachedGraph {
+  CachedGraph(MPSGraph* graph) : MPSCachedGraph(graph) {}
+  MPSGraphTensor* inputTensor_ = nil;
+  MPSGraphTensor* indicesTensor_ = nil;
+  MPSGraphTensor* outputTensor_ = nil;
+};
+
+static std::string getCacheKey(const Tensor& input,
+                               const Tensor& indices,
+                               const ScalarType& dtype,
+                               const IntArrayRef output_size) {
+  return "max_unpooling2d_forward_mps:" + getTensorsStringKey({input, indices}) + "[" + getArrayRefString(output_size) +
```

Comment:
Hi, I'm new to contributing to this project and I don't really grasp yet what to include in cache key? Is output_size and type sufficient? Or is better to include input and indices tensor keys?

Reply:
Cache key should include input and output tensor sizes as well as dtypes. But `getTensorsStringKeys` is probably OK)

Reply:
Great, thanks!